engine/verification/fetcher/engine.go

```diff
@@ -106,352 +102,210 @@ func (e *Engine) Done() <-chan struct{} {
-// a potential processing error internally when done.
-func (e *Engine) Submit(originID flow.Identifier, event interface{}) {
-	e.unit.Launch(func() {
-		err := e.Process(originID, event)
-		if err != nil {
-			engine.LogError(e.log, err)
-		}
-	})
-}
-
-// ProcessLocal processes an event originating on the local node.
-// Note: this method is required as an Engine implementation,
-// however it should not be invoked as match engine requires origin ID of events
-// it receives. Use Process method instead.
-func (e *Engine) ProcessLocal(event interface{}) error {
-	return fmt.Errorf("should not invoke ProcessLocal of Match engine, use Process instead")
-}
-
-// Process processes the given event from the node with the given origin ID in
-// a blocking manner. It returns the potential processing error when done.
-func (e *Engine) Process(originID flow.Identifier, event interface{}) error {
-	return e.unit.Do(func() error {
-		return e.process(originID, event)
-	})
-}
-
-// process receives and submits an event to the engine for processing.
-// It returns an error so the engine will not propagate an event unless
-// it is successfully processed by the engine.
-// The origin ID indicates the node which originally submitted the event to
-// the peer-to-peer network.
-func (e *Engine) process(originID flow.Identifier, event interface{}) error {
-	var err error
-
-	switch resource := event.(type) {
-	case *messages.ChunkDataResponse:
-		err = e.onChunkDataPack(originID, &resource.ChunkDataPack, &resource.Collection)
-	default:
-		return fmt.Errorf("invalid event type (%T)", event)
-	}
-
-	if err != nil {
-		// logs the error instead of returning that.
-		// returning error would be projected at a higher level by network layer.
-		// however, this is an engine-level error, and not network layer error.
-		e.log.Debug().Err(err).Msg("engine could not process event successfully")
-	}
-
-	return nil
-}
-
-// ProcessMyChunk processes the chunk that assigned to me. It should not be blocking since
-// multiple workers might be calling it concurrently.
-// It skips chunks for sealed blocks.
+// ProcessAssignedChunk is the entry point of fetcher engine.
+// It pushes the assigned chunk down the pipeline.
+// Through the pipeline the chunk data pack for this chunk is requested,
+// a verifiable chunk is shaped for it,
+// and is pushed to the verifier engine for verification.
+//
+// It should not be blocking since multiple chunk consumer workers might be calling it concurrently.
 // It fetches the chunk data pack, once received, verifier engine will be verifying
 // Once a chunk has been processed, it will call the processing notifier callback to notify
 // the chunk consumer in order to process the next chunk.
-func (e *Engine) ProcessMyChunk(c *flow.Chunk, resultID flow.Identifier) {
-	chunkID := c.ID()
-	blockID := c.ChunkBody.BlockID
+func (e *Engine) ProcessAssignedChunk(locator *chunks.Locator) {
+	// TODO: add tracing and metrics.
```

Comment:
We address wiring the tracing and metrics in a separate issue for `fetcher` and `requester` engines: https://github.com/dapperlabs/flow-go/issues/5427